src/test/videopress/upload.js

```diff
@@ -0,0 +1,646 @@
+	it( 'displays media options picker when selecting the block', async () => {
+		// Initialize with an empty gallery
+		const {
+			getByLabelText,
+			getByText,
+			getByTestId,
+		} = await initializeEditor( {
+			initialHtml,
+		} );
+
+		fireEvent.press( getByText( 'ADD VIDEO' ) );
+
+		// Observe that media options picker is displayed
+		if ( Platform.isIOS ) {
+			// On iOS the picker is rendered natively, so we have
+			// to check the arguments passed to `ActionSheetIOS`.
+			expect(
+				ActionSheetIOS.showActionSheetWithOptions
+			).toHaveBeenCalledWith(
+				expect.objectContaining( {
+					title: 'Choose video',
+					options: [ 'Cancel', ...MEDIA_OPTIONS ],
+				} ),
+				expect.any( Function )
+			);
+		} else {
+			expect( getByText( 'Choose video' ) ).toBeVisible();
+			MEDIA_OPTIONS.forEach( ( option ) =>
+				expect( getByText( option ) ).toBeVisible()
+			);
+
+			fireEvent( getByTestId( 'media-options-picker' ), 'backdropPress' );
+		}
+
+		// Check that block remains selected after displaying the media
+		// options picker. This is performed by checking if the block
+		// actions menu button is visible.
+		const blockActionsButton = getByLabelText( /Open Block Actions Menu/ );
+		expect( blockActionsButton ).toBeVisible();
+	} );
+
+	it( 'uploads a video from device', async () => {
+		const media = {
+			type: 'video',
+			localId: 1,
+			localUrl: 'file:///local-video-1.mp4',
+			serverId: 1000,
+			serverUrl: 'https://videopress.wordpress.com/local-video-1.mp4',
+		};
+
+		const { notifyUploadingState, notifySucceedState } = setupMediaUpload();
+		const {
+			expectMediaPickerCall,
+			mediaPickerCallback,
+		} = setupMediaPicker();
+
+		const screen = await initializeEditor( {
+			initialHtml,
+		} );
+		const { getByText, getByTestId } = screen;
+		const { selectOption } = setupPicker( screen, MEDIA_OPTIONS );
+		// Clear previous calls to `apiFetch`
+		apiFetch.mockClear();
+
+		// Block is visible
+		const block = await getBlock( screen, 'VideoPress' );
+		expect( block ).toBeVisible();
+
+		// Upload video from device
+		fireEvent.press( getByText( 'ADD VIDEO' ) );
+		selectOption( 'Choose from device' );
+		expectMediaPickerCall( 'DEVICE_MEDIA_LIBRARY', [ 'video' ], false );
+
+		// Block is uploading the video
+		await mediaPickerCallback( media );
+		expect( getByTestId( 'videopress-uploading-video' ) ).toBeVisible();
+		expect( getEditorHtml() ).toMatchSnapshot( 'loading state' );
+
+		// During upload progress we keep displaying the loading state
+		await notifyUploadingState( media );
+		expect( getByTestId( 'videopress-uploading-video' ) ).toBeVisible();
+
+		// Upload finish
+		await notifySucceedState( {
+			...media,
+			metadata: Platform.select( {
+				android: {
+					videopressGUID: VIDEOPRESS_GUID,
+				},
+				ios: {
+					id: VIDEOPRESS_GUID,
+				},
+			} ),
+		} );
+
+		// Requests:
+		//  - Token request
+		//  - Metadata request
+		//  - Check ownership request
```

Comment:
Note this request is relatively new and at some point, we'd need to incorporate the logic in the native version (https://github.com/Automattic/jetpack/pull/30438).